docs/graph_engine/dataset.rst

```diff
@@ -0,0 +1,185 @@
+***********************************
+Ray Dataset and Data Pipeline Usage
+***********************************
+
+In this guide we show how to create and use `Ray Datasets <https://docs.ray.io/en/latest/data/dataset.html>`
+and `Data Pipelines <https://docs.ray.io/en/latest/data/pipelining-compute.html#pipelining-datasets>`
+with the DeepGNN graph engine.
+We show several different sampling strategies and advanced usage.
+
+Generate Dataset
+================
+
+First we generate a Cora dataset and load it into a server to use in our examples.
+We load up a server, when ray tries to pull it into a function, it will pull a
+client instead of the whole server.
+
+.. code-block:: python
+
+    >>> import numpy as np
+    >>> import ray
+    >>> from deepgnn.graph_engine.snark.distributed import Server, Client as DistributedClient
+
+    >>> import tempfile
+    >>> from deepgnn.graph_engine.data.citation import Cora
+    >>> data_dir = tempfile.TemporaryDirectory()
+    >>> Cora(data_dir.name)  # (Train: 140, Valid: 500, Test: 1000)
+    <deepgnn.graph_engine.data.citation.Cora object at 0x...>
+
+    >>> address = "localhost:9999"
+    >>> g = Server(address, data_dir.name, 0, 1)
+
+Simple Cora Dataset
+===================
+
+In this example we create a simple dataset using Ray Data.
+
+First we initialize a Ray Dataset of node ids ranging from 0 to 2708.
+`ray.data.range <https://docs.ray.io/en/latest/data/api/input_output.html#synthetic-data>`
+Then we repartition it to be one block per batch.
+
+.. code-block:: python
+
+    >>> dataset = ray.data.range(2708).repartition(2708 // 512)
+    >>> dataset
+    Dataset(num_blocks=5, num_rows=2708, schema=<class 'int'>)
+
+We convert this dataset to a data pipeline by splitting it into windows.
+
+"Dataset pipelines allow Dataset transformations to be executed incrementally
```

Comment:
Please, avoid copy pasting external documentation and describe what and *why* we are doing in your own words.